src/core/dbus-manager.c

```diff
@@ -1717,6 +1717,10 @@ static int method_soft_reboot(sd_bus_message *message, void *userdata, sd_bus_er
         if (r < 0)
                 return r;
 
+        if (!MANAGER_IS_SYSTEM(m))
```

Comment:
Maybe move it as the first thing? To avoid a needless filesystem check and so on